src/main/java/de/sakpaas/backend/service/SearchService.java

```diff
@@ -74,28 +78,48 @@ public SearchResultObject search(String query, CoordinateDetails coordinateDetai
    * @param coordinateDetails The SearchCoordinates
    * @return A new SearchRequest
    */
+
+
   protected SearchRequest createRequest(String query, CoordinateDetails coordinateDetails)
       throws EmptySearchQueryException {
-    return new SearchRequest();
-  }
+    String lowerquery = query.toLowerCase();
 
-  /**
-   * Possible Brand Names will be extracted from the Query and saved brands List.
-   *
-   * @param request The Request Object
-   * @return the updated Request Object
-   * @throws EmptySearchQueryException Will be thrown if the Query is
-   *                                   Empty(needs to be implemented)
-   */
-  protected SearchRequest checkForBrands(SearchRequest request) {
-    request.setBrands(
-        request.getQuery().stream().filter(queryElement -> knownBrands.contains(queryElement))
-            .collect(
-                Collectors.toSet()));
-    request.getQuery().removeAll(request.getBrands());
-    return request;
-  }
+    SearchRequest searchRequest = new SearchRequest();
+
+
+    ArrayList<Integer> containsbrand = new ArrayList<Integer>();
 
+    HashSet<String> giveQuery = new HashSet<String>();
+    HashSet<String> giveBrands = new HashSet<String>();
+
+
+    for (String brand : knownBrands) {
```

Comment:
Wenn man die Brands in zufälliger Reihenfolge durch geht, dann kann es sein, dass ein kurzer Brandname noch innerhalb des eigentlichen großen Brandnamen gefunden wird. Beispiel aus unserer Datenbank:
`ed` findet man auch in `edeka` wieder.

Um das zu Lösen müsste man mit den längsten Brandnamen anfangen und dann immer kleiner werden, also erst nach `edeka` suchen, dann nach `ed`.